test/unit/strategies/CitrixVDIStrategy.js

```diff
@@ -0,0 +1,81 @@
+import sinon, { sandbox } from 'sinon';
+import chai from 'chai';
+import CitrixVDIStrategy from '../../../src/js/strategies/CitrixVDIStrategy.js';
+
+describe('CitrixVDIStrategy', () => {
+    beforeEach(() => {
+        global.window = {
+            CitrixWebRTC: {
+                setVMEventCallback: sandbox.stub(),
+                isFeatureOn: sandbox.stub(),
+                initUCSDK: sandbox.stub(),
+                initLog: sandbox.stub(),
+            },
+            getCitrixWebrtcRedir: null,
+        };
+        global.connect = { getLog: sandbox.stub() };
+    });
+
+    afterEach(() => {
+        sandbox.restore();
+        delete global.window;
+        delete global.connect;
+    });
+
+    describe('constructor', () => {
+        it('should initialize CitrixWebRTC, getCitrixWebrtcRedir, and logging', () => {
+            const instance = new CitrixVDIStrategy(false);
+
+            chai.expect(global.window.CitrixWebRTC.setVMEventCallback).to.have.been.calledOnce;
+            chai.expect(global.window.getCitrixWebrtcRedir).to.be.a('function');
```

Comment:
is it possible to have a stronger assertion?

Reply:
Citrix need this `window.getCitrixWebrtcRedir ` to be a function, with return value being a promise of value 0, 1, or empty